pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -404,8 +405,11 @@ func handleModify(ctx *downloaderContext, key string,
 		status.IsContainer = config.IsContainer
 		publishDownloaderStatus(ctx, status)
 	}
-	// If RefCount from zero to non-zero then do install
-	if status.RefCount == 0 && config.RefCount != 0 {
+	if status.PendingAdd {
```

Comment:
PendingAdd should be cleared in all cases when handleCreate returns. Is it not?
If a download is running for a particular Key, then the goroutine is presumably busy hence a handleModify would not be processed. What am I missing?

Reply:
PendingAdd will be cleared once the download completes or handleCreate returns. But if the download is in progress and we deployed the second instance then it will again call the handleCreate.
Or we can do one more thing, we will trigger handleCreate from handleModify only if status got expired then also our problem will get fixed. Let me update the PR.

Reply:
> If a download is running for a particular Key, then the goroutine is presumably busy hence a handleModify would not be processed. What am I missing?

handleModify gets processed in this case. I verified it in my testing.
Won't it spawn a new goroutine for a new call of handle create from handleModify?

Reply:
Are you talking about a second download of the same Key and the same bits?

Reply:
The goroutine for that Key would be busy running the download AFAIU